record-and-playback/core/scripts/rap-process-worker.rb

```diff
@@ -23,11 +23,18 @@
 require 'yaml'
 require 'fileutils'
 
+
+# This function handle file sort by timestamp @dseral
+def sort_by_date(arr)
+        h = Hash[arr.collect { |x| [ x.chomp(".done").split("-")[1], x ] }].sort
+        return hh=h.map { |key, value| value }
+end
```

Comment:
Can I suggest rewriting this method to remove unnecessary complexity?
```suggestion
def sort_by_date(filenames)
  filenames.sort_by { |filename| filename.chomp(".done").split("-").second }
end
```

Reply:
Ubuntu 16.04 come with ruby 2.3.1 and unfortunately .second method is available since version 4.02. Do you have another suggestion?

Reply:
Apologies, I'm used to work with Ruby in the context of Rails. I have updated the suggestion.

Reply:
It seems working.. Thanks louim..